controllers/runner_controller.go

```diff
@@ -671,10 +671,29 @@ func (r *RunnerReconciler) newPod(runner v1alpha1.Runner) (corev1.Pod, error) {
 	runnerSpec := runner.Spec
 
 	if len(runnerSpec.VolumeMounts) != 0 {
+		// if operater provides a work volume mount, use that
```

Comment:
An alternative to stripping/removing the work VolumeMount here is simply adding it here if it's not provided by the user.
From reading the code, the functions where it's added should be like a "default" template, so I opted for this.

I tried skipping it in that default template, but the needed context is not always present.
The user provided volumes is not available, but easily refactored. Only issue I couldn't overcome was the "registration-only" codepath was somehow different.

Reply:
Once https://github.com/actions-runner-controller/actions-runner-controller/pull/1028 is merged I can simplify this pr and make it cleaner/neater.